Cargo.toml

```diff
@@ -56,13 +56,13 @@ memmap = { package = "memmap2", version = "0.9" }
 multiversion = "0.7"
 ndarray = { version = "0.16", default-features = false }
 num-traits = "0.2"
-numpy = "0.22"
+numpy = "0.23"
 object_store = { version = "0.11", default-features = false }
 once_cell = "1"
 parking_lot = "0.12"
 percent-encoding = "2.3"
 pin-project-lite = "0.2"
-pyo3 = "0.22"
+pyo3 = { git = "https://github.com/bschoenmaeckers/pyo3.git", branch = "release-0.23" }
```

Comment:
I can add back the temporary fix for https://github.com/PyO3/pyo3/issues/4764 until https://github.com/PyO3/pyo3/pull/4766 is released. If that is desirable.